test/SlowTests/Issues/RavenDB-14978.cs

```diff
@@ -38,21 +38,47 @@ public async Task Can_setup_write_load_balancing()
                 }
             }.Initialize();
 
-            var amre = new AsyncManualResetEvent();
+            var (index, _) = await CreateDatabaseInCluster(databaseName, 3, leader.WebUrl);
+            await WaitForRaftIndexToBeAppliedInCluster(index, TimeSpan.FromSeconds(30));
 
-            var requestExecutor = store.GetRequestExecutor();
-            requestExecutor.ClientConfigurationChanged += (sender, _) => amre.Set();
+            Can_setup_write_load_balancing(store, ref context);
+        }
+
+        [Fact]
+        public async Task Can_setup_write_load_balancing_on_server()
+        {
+            var leader = await CreateRaftClusterAndGetLeader(3);
+            var databaseName = GetDatabaseName();
+            var context = "users/1";
+
+            using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(60));
+            using var store = new DocumentStore
+            {
+                Urls = new[] { leader.WebUrl },
+                Database = databaseName,
+                Conventions = new DocumentConventions
+                {
+                    LoadBalancerPerSessionContextSelector = db => context
+                }
+            }.Initialize();
+
+            var configuration = new ClientConfiguration
+            {
+                ReadBalanceBehavior = ReadBalanceBehavior.RoundRobin, 
+                LoadBalanceBehavior = LoadBalanceBehavior.UseSessionContext, 
+                Disabled = false
+            };
 
             var (index, _) = await CreateDatabaseInCluster(databaseName, 3, leader.WebUrl);
             await WaitForRaftIndexToBeAppliedInCluster(index, TimeSpan.FromSeconds(30));
 
-            using (var s0 = store.OpenSession())
-            {
-                s0.Load<User>("test/1");
-            }
+            await store.Maintenance.SendAsync(new PutClientConfigurationOperation(configuration), cts.Token);
```

Comment:
where are you wait to be notified about this?

Reply:
I am testing the behavior of `LoadBalanceBehavior.UseSessionContext`.
And there is no need to wait while in the new implementation the server sends in the header of `PutClientConfigurationOperation` request the need to update so the client does it immediately after this request

Reply:
The update is done in the background, no? What happens if this takes too long?
That would make this a racy test, no?

Reply:
We do await for both the client configuration update task and the topology update task
https://github.com/ravendb/ravendb/blob/3d7a25151b978e717edd9f3fab2e965b085b4a87/src/Raven.Client/Http/RequestExecutor.cs#L897

Reply:
Okay, then LGTM